tests/otel_integrations/test_celery.py

```diff
@@ -0,0 +1,106 @@
+"""This module contains tests for the Celery integration.
+
+To be able to run those tests, you need to have Redis running on localhost:6379.
+We have a docker compose file that you can use to start Redis:
+
+    ```bash
+    docker compose up -d redis
+    ```
+"""
+
+import logging
+from typing import Iterator
+
+import pytest
+import redis
+import redis.exceptions
+from celery import Celery
+from celery.contrib.testing.worker import start_worker
+from celery.worker.worker import WorkController
+from dirty_equals import IsStr
+from inline_snapshot import snapshot
+from opentelemetry.instrumentation.celery import CeleryInstrumentor
+
+import logfire
+from logfire.testing import TestExporter
+
+pytestmark = [pytest.mark.integration]
+
+try:
+    client = redis.Redis()
+    client.ping()  # type: ignore
+except redis.exceptions.ConnectionError:
+    pytestmark.append(pytest.mark.skip('Redis is not running'))
+
+
+@pytest.fixture
+def celery_app() -> Iterator[Celery]:
+    app = Celery('tasks', broker='redis://localhost:6379/0', backend='redis://localhost:6379/0')
+
+    @app.task(name='tasks.say_hello')  # type: ignore
+    def say_hello():  # type: ignore
+        return 'hello'
+
+    logfire.instrument_celery()
+    try:
+        yield app
+    finally:
+        CeleryInstrumentor().uninstrument()  # type: ignore
+
+
+@pytest.fixture(autouse=True)
+def celery_worker(celery_app: Celery) -> Iterator[WorkController]:
+    logger = logging.getLogger()
+    with start_worker(celery_app, perform_ping_check=False, loglevel=logger.level) as worker:  # type: ignore
```

Comment:
The `loglevel` has a default value to `ERROR`, and it modifies the root logger.

This code just sets the current level.

Reply:
This comment should be in the code